chia/types/blockchain_format/program.py

```diff
@@ -246,7 +246,7 @@ def run_with_cost(self, max_cost: int, *args) -> Tuple[int, Program]:
 
     # returns an optional error code and an optional PySpendBundleConditions (from clvm_rs)
     # exactly one of those will hold a value
-    def run_as_generator(self, max_cost: int, flags: int, *args) -> Tuple[Optional[uint16], Optional[Any]]:
+    def run_as_generator(self, max_cost: int, flags: int, *args) -> Union[Tuple[None, Any], Tuple[uint16, None]]:
```

Comment:
@arvidn, is this hint a more accurate representation of what `clvm_rs.run_generator2()` returns?  I think both the Python hints and whatever Rust calls them (I swear I'll learn Rust someday...) are accurate but overly broad compared to the two things that are actually returned.

https://github.com/Chia-Network/clvm_rs/blob/781a74438a9c4c1168b285b9fbbcb6ebd466e931/src/py/run_generator.rs#L145
```rust
) -> PyResult<(Option<ErrorCode>, Option<PySpendBundleConditions>)> {
```
https://github.com/Chia-Network/clvm_rs/blob/781a74438a9c4c1168b285b9fbbcb6ebd466e931/src/py/run_generator.rs#L170-L187
```rust
    match r {
        Ok((None, Some(spend_bundle_conds))) => {
            // everything was successful
            Ok((
                None,
                Some(convert_spend_bundle_conds(
                    py,
                    &allocator,
                    spend_bundle_conds,
                )),
            ))
        }
        Ok((error_code, _)) => {
            // a validation error occurred
            Ok((error_code, None))
        }
        Err(eval_err) => eval_err_to_pyresult(py, eval_err, allocator),
    }
```

Reply:
I suppose so. the `Any` was really just me being lazy trying to find out how to refer to the rust type.